error.go

```diff
@@ -75,7 +105,12 @@ func flushErrors() {
 func fetchError() error {
 	select {
 	case err := <-lastError:
-		return err
+		switch err.Code {
+		case outOfMemory, invalidEnum, invalidValue, notInitialized, noCurrentContext:
```

Comment:
It seems a little odd these are not in the same order as above, but I guess that's fine.